source/module_base/test/math_sphbes_test.cpp

```diff
@@ -0,0 +1,181 @@
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l0,jl);
+    //reference result is from bessel_test.cpp which is calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1
+    EXPECT_NEAR(mean(jl,msh)/0.2084468748396,   1.0,doublethreshold); 
+
+
+    //int l = 1;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l1,jl);
+    //reference result is from bessel_test.cpp which is calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1
+    EXPECT_NEAR(mean(jl,msh)/0.12951635180384,   1.0,doublethreshold);
+
+
+    //int l = 2;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l2,jl);
+    //the result from bessel_test.cpp calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1 is 0.124201140093879
+    //reference result is calculated by Sphbes::Spherical_Bessel(msh,r,q,l,jl)
+    EXPECT_NEAR(mean(jl,msh)/0.12420114009271901456,   1.0,doublethreshold);
+
+    //int l = 3;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l3,jl);
+    //the result from bessel_test.cpp calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1 is 0.118268654505568
+    //reference result is calculated by Sphbes::Spherical_Bessel(msh,r,q,l,jl)
+    EXPECT_NEAR(mean(jl,msh)/0.11826865448477598408,   1.0,doublethreshold);
+
+
+    //int l = 4;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l4,jl);
+    //the result from bessel_test.cpp calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1 is 0.0933871035384385
+    //reference result is calculated by Sphbes::Spherical_Bessel(msh,r,q,l,jl)
+    EXPECT_NEAR(mean(jl,msh)/0.093387100084701621383,   1.0,doublethreshold);
+
+
+    //int l = 5;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l5,jl);
+    //the result from bessel_test.cpp calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1 is 0.0603800487910689
+    //reference result is calculated by Sphbes::Spherical_Bessel(msh,r,q,l,jl)
+    EXPECT_NEAR(mean(jl,msh)/0.060380048719821471925,   1.0,doublethreshold);
+
+
+    //int l = 6;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l6,jl);
+    //the result from bessel_test.cpp calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1 is 0.0327117051555907
+    //reference result is calculated by Sphbes::Spherical_Bessel(msh,r,q,l,jl)
+    EXPECT_NEAR(mean(jl,msh)/0.032711705053977857549,   1.0,doublethreshold);
+
+
+    //int l = 7;
+    ModuleBase::Sphbes::Spherical_Bessel(msh,r,q,l7,jl);
+    //the result from bessel_test.cpp calculated by
+    //ModuleBase::Sph_Bessel_Recursive::D1 is 0.0152155566653926
+    //reference result is calculated by Sphbes::Spherical_Bessel(msh,r,q,l,jl)
+    EXPECT_NEAR(mean(jl,msh)/0.015215556095798710851,   1.0,doublethreshold);
+}
+
+TEST_F(Sphbes,SphericalBesselRoots)
+{
+    int neign = 100;
+    double **eign = new double*[8];
+    for(int i=0;i<8;++i)
+    {
+        eign[i] = new double[neign];
+        ModuleBase::Sphbes::Spherical_Bessel_Roots(neign,i,1.0e-12,eign[i],10.0);
+    }
+
+    EXPECT_NEAR(eign[0][0]/0.31415926535899563188,       1.0,doublethreshold);
```

Comment:
Why are you using expect_near with (calculated/expected,1), rather than (calculated, expected)?

Reply:
I have a series of testes, and the values may have different orders of magnitude, so it is not convenient to use a same threshold. While, by using this type, I can use the same threshold.

Reply:
> I have a series of testes, and the values may have different orders of magnitude, so it is not convenient to use a same threshold. While, by using this type, I can use the same threshold.

OK, I know. Personally, I prefer the way of `expect_near(calculated,expected, expected * error)`. It's all up to you.

Reply:
Ok, I think the current form is good.